### PR TITLE
dev-libs/uriparser: bump to 0.8.5. Closes #2195

### LINKED ISSUE #2195: Please update to uriparser 0.8.5
Hi!

uriparser 0.8.5 has been released:
https://github.com/uriparser/uriparser/releases

Thanks for updating!

### PR DESCRIPTION
**Pull request by fbrosson**
* Add support for secondary arch.
* Fix `PROVIDES_devel`.
* Add debuginfo package.

### CODE REVIEW

File: dev-libs/uriparser/uriparser-0.8.5.recipe [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+libVersion="1.0.21"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	uriparser$secondaryArchSuffix = $portVersion
+	lib:liburiparser$secondaryArchSuffix = $libVersionCompat
+	"
+if [ -z "$secondaryArchSuffix" ]; then
+PROVIDES="$PROVIDES
+	cmd:uriparse$secondaryArchSuffix = $portVersion
```
[the remark this answers is not in the document]
Comment (fbrosson) [reply dev-libs/uriparser/uriparser-0.8.5.recipe]: Good catch, thanks!
(Hopefully this is not an error, so we can remove the `$secondaryArchSuffix` next time there is a version bump.)

File: dev-libs/uriparser/uriparser-0.8.5.recipe [hunk 2]
```diff
@@ -0,0 +1,81 @@
+SUMMARY="A strictly RFC 3986 compliant URI parsing library in C"
+DESCRIPTION="Uriparser is a strictly RFC 3986 compliant URI parsing library \
+written in C."
+HOMEPAGE="https://uriparser.github.io/"
+COPYRIGHT="2007 Weijia Song
+	2007-2018 Sebastian Pipping"
```
[the remark this answers is not in the document]
Comment (fbrosson) [reply dev-libs/uriparser/uriparser-0.8.5.recipe]: Thanks, I'll have a look at it.

File: dev-libs/uriparser/uriparser-0.8.5.recipe [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+defineDebugInfoPackage uriparser$secondaryArchSuffix \
+	"$libDir"/liburiparser.so.$libVersion
+
+BUILD()
+{
+	libtoolize --force --copy --install
+	aclocal -I m4
+	autoconf
+	automake --add-missing
```
[the remark this answers is not in the document]
Comment (fbrosson) [reply dev-libs/uriparser/uriparser-0.8.5.recipe]: OK, thanks.
Actually, I just kept what was in the recipe of the previous version.
(I'll change this to `autoreconf` next time there is a version bump or if I fix `COPYRIGHT`.)

**Comment by hartwork** [comment]
@fbrosson thanks!

**Pull request merged**